### PR TITLE
code cleanup before server-server margo

### PR DESCRIPTION
**Pull request by MichaelBrim**
- Fixes build warnings for gcc and XL compilers
- Moves files into/out of common where appropriate
- Cleans up #includes for client and server
- Remove --enable-debug configure option

Note on '--enable-debug': other than modifying user CFLAGS, which
is considered bad practice, we were not doing anything useful when
this option was given. The recommended method to support debug
builds is to have the user pass the appropriate CFLAGS on the
configure command.

### Types of changes

- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Performance enhancement (non-breaking change which improves efficiency)
- [x] Code cleanup (non-breaking change which makes code smaller or more readable)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)
- [ ] Documentation (a change to man pages or other documentation)

### Checklist:

- [x] My code follows the UnifyCR code style requirements.
- [ ] I have updated the documentation accordingly.
- [x] I have read the **CONTRIBUTING** document.
- [ ] I have added tests to cover my changes.
- [x] All new and existing tests passed.
- [x] All commit messages are properly formatted.

### CODE REVIEW

**Comment by CamStan** [comment]
No idea if this is the case, but if since Travis is building off of the cache, any changes to the way the build goes can make things go weird. This environment might now have mercury, mercury@develop, margo, and margo@develop. Might need to source the one you want since you changed it in the travis.yml, and maybe clearing the cache will help.

**Comment by MichaelBrim** [comment]
I should note that on Summitdev, with the most recent mercury spackage, I had to install:
```
mercury@develop+sm
margo@develop ^mercury@develop+sm
```

**Comment by CamStan** [comment]
> I should note that on Summitdev, with the most recent mercury spackage, I had to install:
> 
> ```
> mercury@develop+sm
> margo@develop ^mercury@develop+sm
> ```
> 
> This is because the new spackage turns off the shared-memory plugin by default, but we use it in Unify.

I imagine we'll need to adjust our Spack package for this then?

**Comment by MichaelBrim** [comment]
@CamStan yes, since the updated sds-repo spackages got upstreamed 12 days ago.

**Comment by MichaelBrim** [comment]
@CamStan we may also want to eliminate using the sds-repo in the Travis build since the packages are all in upstream

**Comment by CamStan** [comment]
Oh, nice! I'll get on it.

File: configure.ac [hunk 1]
```diff
@@ -87,20 +87,6 @@ else
 AM_CONDITIONAL([USE_PMIX], [false])
 fi
 
-# Debug build
-AC_ARG_ENABLE([debug],
-              AC_HELP_STRING([--enable-debug],
-                             [Enable debug build options.]))
```
Comment (CamStan) [inline configure.ac]: I'm not clear on why the removal of this option. Currently updating our spackage which has a variant for this.

Comment (MichaelBrim) [reply configure.ac]: It wasn't doing anything useful - see note in the issue description. If someone wants a debug build, they should use `configure <config-opts> "CFLAGS=-g -O0"` or similar

Comment (MichaelBrim) [reply configure.ac]: The actual CFLAGS to pass are compiler dependent.

**Pull request merged**